src/lib/annotations/AnnotationThread.js

```diff
@@ -397,6 +378,38 @@ class AnnotationThread extends EventEmitter {
     //--------------------------------------------------------------------------
 
     /**
+     * Update a temporary annotation with the annotation saved on the backend and set the threadNumber if it does
+     * not exist. Propogate the threadnumber to dialog if applicable.
+     *
+     * @private
+     * @param {Annotation} serverAnnotation - The annotation returned by the backend. Use this as the source of truth
+     * @param {Annotation} localAnnotation - The temporary annotation holding place of the server validated annotation
+     * @return {void}
+     */
+    updateLocalAnnotationFromServer(serverAnnotation, localAnnotation) {
+        const tempIdx = this.annotations.indexOf(localAnnotation);
+        if (tempIdx === -1) {
+            // If no temporary annotation is found, save to thread normally
+            this.saveAnnotationToThread(serverAnnotation);
+        } else {
+            // Otherwise, replace temporary annotation with annotation saved to server
+            this.annotations[tempIdx] = serverAnnotation;
+        }
+
+        this.threadNumber = this.threadNumber || serverAnnotation.threadNumber;
+
+        if (this.dialog) {
+            // Add thread number to associated dialog and thread
+            if (this.dialog.element && this.dialog.element.dataset) {
+                this.dialog.element.dataset.threadNumber = this.threadNumber;
+            }
+
+            this.dialog.addAnnotation(serverAnnotation);
```

Comment:
// Removes temporary/local annotation and replaces it with the annotation saved to server